Modules/gcmodule.c

```diff
@@ -1062,20 +1062,38 @@ delete_garbage(PyThreadState *tstate, GCState *gcstate,
     }
 }
 
+static void
+clear_freelists(PyFreeListState *state)
+{
+    _PyList_ClearFreeList(state);
+}
+
 /* Clear all free lists
  * All free lists are cleared during the collection of the highest generation.
  * Allocated items in the free list may keep a pymalloc arena occupied.
  * Clearing the free lists may give back memory to the OS earlier.
  */
 static void
-clear_freelists(PyInterpreterState *interp)
+clear_all_freelists(PyInterpreterState *interp)
 {
     _PyTuple_ClearFreeList(interp);
     _PyFloat_ClearFreeList(interp);
-    _PyList_ClearFreeList(interp);
     _PyDict_ClearFreeList(interp);
     _PyAsyncGen_ClearFreeLists(interp);
     _PyContext_ClearFreeList(interp);
+#if defined(Py_GIL_DISABLED)
```

Comment:
@colesbury
Not related to this PR, but can we make each thread clear its own freelists concurrently in the near future?

Reply:
Yeah, that might make sense in free-threaded builds.

Reply:
I will create a issue for this 👍